libraries/AC_PrecLand/AC_PrecLand_Companion.h

```diff
@@ -44,5 +78,21 @@ class AC_PrecLand_Companion : public AC_PrecLand_Backend
     Vector3f            _los_meas_body;         // unit vector in body frame pointing towards target
     bool                _have_los_meas;         // true if there is a valid measurement from the camera
     uint32_t            _los_meas_time_ms;      // system time in milliseconds when los was measured
-    bool                _wrong_frame_msg_sent;
+
+    Vector3f            _los_meas_local_frd;            // FRD local  frame (x: Front, y: Right, z: Down) with origin that travels with the vehicle Z - down to earth.
+    bool                _have_los_meas_local_frd;       // true if there is a valid FRD data
+    uint32_t            _los_meas_local_frd_time_ms;    // system time in milliseconds when los was measured
```

Comment:
There are quite a few bytes of RAM going missing in here - we'll only use 1/3 of these at any one point.

Perhaps we could instead:
```
Vector3f _los_meas;
MAV_FRAME _los_meas_frame;
MAV_FRAME _last_los_meas_frame;
uint32_t _last_ros_meas_time_ms;
```

Reply:
@peterbarker , yes, in general, this PR looks to be updated - regarding the idea from @rishabsingh3003 to move all frames conversion and all code to backend.cpp. So it needs to be wholly rebuilt PR with the new concept. I hope to find time asap to do it.